### ISSUE TITLE
Cleary identify 'Islandora' content types in the UI

### ISSUE THREAD

**Issue by dannylamb** [Islandora/documentation#1255]
Since a user can make any content type, and that content type need not be "Islandora compatible", then it would be beneficial to highlight what content types / media types / taxonomy vocabularies are meant to work with Islandora.

**Comment by mjordan** [comment]
Related issue: [Members and Media tabs appear on non-Repository Item pages](https://github.com/Islandora-CLAW/CLAW/issues/1057)

**Comment by dannylamb** [comment]
Here's some mock ups from @rosiel 

![mockup-content_types](https://user-images.githubusercontent.com/20773151/63945546-df57b000-ca49-11e9-8486-178815c2047a.png)

![mockup-add_content](https://user-images.githubusercontent.com/20773151/63945553-e2eb3700-ca49-11e9-84a1-a7e3fabfc3a3.png)

**Comment by rosiel** [comment]
At the moment, the following fields are "necessary":
Nodes:
- `field_member_of`
- `field_model`
Media:
- `field_media_of`
- `field_media_use`
Taxonomy:
- `field_external_uri`

Do we want to bake this in? Create a function that checks whether these exact fields are present? (seems very do-able, and/but hard-codey)

To make it less hard-codey, should that function include an alter or some other way for modules to plug in, so i can write a custom module for my repo that checks additional conditions? Challenge: create logic table that works sensibly for that, if "core" and (N different) "custom" modules disagree.

Or, Let the user decide. Put a checkbox on the Content Type itself, and it doesn't matter what fields exist. Unfortunately, basic islandora functionality might not work on things that are "marked" islandora but don't meet the basic criteria.

Or, let the user check a checkbox but only make that checkbox available/relevant when required fields are present. I.e. to avoid a case where the user calls it islandora then deletes required fields, the test that "this is islandora()" should check for required fields first and then the value of the checkbox. When the checkbox is off and disabled because the required fields aren't present, help text should explain what is missing.

I kind of like the last one, what do y'all think? 

[oops, edited cause I forgot to remove some draft text]

**Comment by dannylamb** [comment]
I would just have the function check for those fields and then use that function everywhere.  If people need additional conditions we can add the hook but that's not necessary at this point in time. People already can choose when/how things go to Fedora using checkboxes, and there's clearly been backlash against that flexibility, so let's not provide more checkboxes.

**Comment by seth-shaw-unlv** [comment]
What makes something an "Islandora" content type or media is dependent on the enabled contexts; that is where the "flush this to Fedora" checkbox for a content type resides.  So, perhaps the check is to see whether or not that content type has the relevant contexts configured/enabled. 

That said, it would be useful to have an "Islandora-ize" operation in the content type operations dropdown to automate all the configuration that needs to happen (adding the fields and updating the appropriate contexts).

**Comment by dannylamb** [comment]
Maybe I didn't give the best example, as 'being in Fedora' does not necessarily make 'Islandora content'.  So I'd definitely scope this to checking for those fields, since that's what makes content compatible with our derivative and collection management methodologies.  I guess that's what Islandora content really is.

Islandora-izing would _definitely_ be useful.  There's lot of utility there.  You model your content, hit a button, and then we add the appropriate fields automagically.  I like that.

**Comment by rosiel** [comment]
> it would be useful to have an "Islandora-ize" operation in the content type operations dropdown to automate all the configuration that needs to happen (adding the fields and updating the appropriate contexts).

Yes! But I agree with Danny - fields yes, contexts have to remain configurable. (there is no specific behaviour that makes something "Islandora".) 

A future ... thing? could be making a Context condition that uses this function. Imagine a context that said: IF (node type is islandora) THEN (flush to fedora)... which (for the things we want to do to ALL islandora stuff) would be easier than having to check off individual content types.

**Comment by mjordan** [comment]
A Context Condition that checked whether a node is islandora would be very useful. :+1:

**Comment by alexkent0** [comment]
Hi all, after reading Rosie's post and the thread, my mind went to using
Context somehow to automatically tag if that media type is Islandora vs
Drupal. This would help make the workflow smooth. Thought I would just
confirm that this seems like a great idea and improvement. It's been really
interesting reading the thread today.
